src/audio/base_fw.c

```diff
@@ -467,6 +467,45 @@ static int global_perf_data_get(uint32_t *data_off_size, char *data)
 #endif
 }
 
+static int io_global_perf_state_get(uint32_t *data_off_size, char *data)
+{
+#ifdef CONFIG_SOF_TELEMETRY_IO_PERFORMANCE_MEASUREMENTS
+	*data = io_perf_monitor_get_state();
+	*data_off_size = sizeof(enum ipc4_perf_measurements_state_set);
+
+	return IPC4_SUCCESS;
+#else
+	return IPC4_UNAVAILABLE;
+#endif
+}
+
+static int io_global_perf_data_get(uint32_t *data_off_size, char *data)
```

Comment:
why `char`? what does it have to do with characters / strings?

Reply:
A better question would be: why all the other handlers have this kind of interface